mongodbatlas/performance_advisor.go

```diff
@@ -0,0 +1,200 @@
+}
+
+// Stats represents query statistics.
+type Stats struct {
+	MS        float64 `json:"ms,omitempty"`        // Duration in milliseconds of the query.
+	NReturned int64   `json:"nReturned,omitempty"` // Number of results returned by the query.
+	NScanned  int64   `json:"nScanned,omitempty"`  // Number of documents read by the query.
+	TS        int64   `json:"ts,omitempty"`        // Query timestamp, in seconds since epoch.
+}
+
+// SuggestedIndex represents a suggested index.
+type SuggestedIndex struct {
+	ID        string           `json:"id,omitempty"`        // Unique id for this suggested index.
+	Impact    []string         `json:"impact,omitempty"`    // List of unique identifiers which correspond the query shapes in this response which pertain to this suggested index.
+	Namespace string           `json:"namespace,omitempty"` // 	Namespace of the suggested index.
+	Weight    float64          `json:"weight,omitempty"`    // Estimated percentage performance improvement that the suggested index would provide.
+	Index     []map[string]int `json:"index,omitempty"`     // Array of documents that specifies a key in the index and its sort order, ascending or descending.
+}
+
+// SuggestedIndexes represents an array of suggested indexes.
+type SuggestedIndexes struct {
+	SuggestedIndexes []*SuggestedIndex `json:"suggestedIndexes,omitempty"` // Documents with information about the indexes suggested by the Performance Advisor.
+	Shapes           []*Shape          `json:"shapes,omitempty"`           // Documents with information about the query shapes that are served by the suggested indexes.
+
+}
+
+// PerformanceAdvisorRequest contains the request query parameters for the API request.
+type PerformanceAdvisorRequest struct {
+	Since      int64  `json:"since,omitempty"`      // Point in time, specified as milliseconds since the Unix Epoch, from which you want to receive results.
+	Duration   int64  `json:"duration,omitempty"`   // 	Length of time from the since parameter, in milliseconds, for which you want to receive results.
+	Envelope   bool   `json:"envelope,omitempty"`   // Boolean that specifies whether or not to wrap the response in an envelope. Default is false.
+	Namespaces string `json:"namespaces,omitempty"` // Namespaces from which to retrieve slow query logs. A namespace consists of the database and collection resource separated by a ., such as <database>.<collection>.
+	NLogs      int64  `json:"nLogs,omitempty"`      // Maximum number of log lines to return. Defaults to 20000.
+	Pretty     bool   `json:"pretty,omitempty"`     // Display response in a prettyprint format. Defaults to false.
+	NIndexes   int64  `json:"nIndexes,omitempty"`   // Maximum number of indexes to suggest. Defaults to unlimited.
+	NExamples  int64  `json:"NExamples,omitempty"`  // Maximum number of examples queries to provide that will be improved by a suggested index. Defaults to 5.
```

Comment:
I already suggested renaming this to query or maybe options but I'd also split this into 3 structures given that not all fields are available to all three methods `NIndexes` and `NExamples` are Indexes only `nLogs` is slow query only, this may not be a problem as omit empty will solve not setting it but what if I set it? or if me as a user think I can set it just to be ignore? so my comment is more about usability than anything else, we should avoid users being confused by the usage of this struct and as such we should offer just what they can set

Finally given that this is for query params don't use the `json` tag, it should be `url`